src/pages/KadraVykhovnykiv/AddNewKadraForm.tsx

```diff
@@ -259,22 +263,31 @@ type FormAddKadraProps = {
                   message: maxNumber(99999),
                 },
                 {
-                  validator: async (_ : object, value: number) =>
+                  validator: async (_ : object, value: any) =>
+                  value ?
+                    !isNaN(value) ?
                       value < 1
                           ? Promise.reject(minNumber(1)) 
                           : await KadraVykhovnykivApi
                               .doesRegisterNumberExist(value)
                               .then(response => response.data === false)
                               ? Promise.resolve()
                               : Promise.reject('Цей номер уже зайнятий')
+                              : Promise.reject()
+                              : Promise.reject()
                 }
-              ]}
-          >
+              ]}>
             <Input
-              type="number"
```

Comment:
Doesn't type="number" preventing us to type there anything except number, does it?